rasa/shared/core/training_data/visualization.py

```diff
@@ -473,11 +473,18 @@ def visualize_neighborhood(
                 and event_idx.action_name == ACTION_LISTEN_NAME
             ):
                 next_node_idx += 1
+                if message is None:
+                    label = "  ?  "
+                else:
+                    intent = message.get("intent", {})
+                    label = (
+                        intent.get("name", "  ?  ")
+                        if isinstance(intent, dict)
```

Comment:
what would be the type of `intent` if not a dict?

Reply:
mypy seems to believe it can be also of type `object`, which is why I had to introduce this awkward if.